### PR TITLE
Add cursor CSS property

### LINKED ISSUE #92: Add `cursor` CSS property
Reference: https://developer.mozilla.org/en-US/docs/Web/CSS/cursor
In Clay: https://github.com/sebastiaanvisser/clay/blob/master/src/Clay/Display.hs#L256.

### PR DESCRIPTION
**Pull request by vyorkin**
Adds is a very basic cursor support.
Cursor URLs aren't implemented.

Resolves #92

### CODE REVIEW

**Comment by vyorkin** [comment]
Please, don't merge yet, going to refactor this to replace `newtype` with sum type like we did in #95

**Comment by vyorkin** [comment]
Refactored, ready for review

File: src/CSS/Cursor.purs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  | EResize
+  | SResize
+  | WResize
+  | NEResize
+  | NWResize
+  | SEResize
+  | SWResize
+  | EWResize
+  | NSResize
+  | NESWResize
```
Comment (vyorkin) [inline src/CSS/Cursor.purs]: not a best name, but that's how its named in CSS :)

**Comment by cprussin** [comment]
Any chance of this getting rebased and merged @vyorkin or @garyb ?  If you guys need, I'm happy to handle the rebase

**Comment by vyorkin** [comment]
btw, there is a couple of other pending PR's as well (which are already rebased)

**Comment by milesfrain** [comment]
@thomashoneyman Would you merge this if we do another rebase?

**Comment by thomashoneyman** [comment]
Yes, I will. Thanks everyone!

**Comment by thomashoneyman** [comment]
@vyorkin Happy to merge this once it's conflict-free. Thanks!

**Pull request merged**